src/core/js/views/adaptView.js

```diff
@@ -139,20 +133,25 @@ define([
       }
     }
 
-    preRemove() {}
+    preRemove() {
+      const type = this.constructor.type;
+      Adapt.trigger(`${type}View:preRemove view:preRemove`, this);
+    }
 
     remove() {
-
+      const type = this.constructor.type;
       this.preRemove();
+      Adapt.trigger(`${type}View:remove view:remove`, this);
       this._isRemoved = true;
 
       Adapt.wait.for(end => {
-
         this.$el.off('onscreen.adaptView');
         this.model.setOnChildren('_isReady', false);
         this.model.set('_isReady', false);
-        Backbone.View.prototype.remove.call(this);
-
+        super.remove();
+        _.defer(() => {
+          Adapt.trigger(`${type}View:postRemove view:postRemove`, this);
```

Comment:
lazy tracking of leaving a page or lazy saving of model states etc. these would then be done async off the back of the event loop because of the defer rather than sync in the main rendering process. general stuff like that, cleanup that can be done later.